src/flagpole/evaluation_context.py

```diff
@@ -13,40 +13,78 @@
 EvaluationContextDict = dict[str, ValidContextTypes]
 
 
+class InvalidIdentityFieldException(Exception):
+    pass
+
+
 class EvaluationContext:
     """
     Prepared by the application and passed to flagpole to evaluate
     feature conditions.
     """
 
-    def __init__(self, data: EvaluationContextDict):
+    __data: EvaluationContextDict
+    __identity_fields: set[str]
+    id: int
+
+    def __init__(self, data: EvaluationContextDict, identity_fields: set[str] | None = None):
         self.__data = deepcopy(data)
 
-    def get(self, key: str) -> Any:
-        return self.__data.get(key)
+        if identity_fields is None:
+            # If no list of identity fields is provided, use all properties of
+            # the context.
+            self.__identity_fields = set(self.__data.keys())
+        else:
+            self.__identity_fields = deepcopy(identity_fields)
+            self.__validate_identity_fields()
 
-    def has(self, key: str) -> Any:
-        return key in self.__data
+        self.__id = self.__generate_id()
 
-    def size(self) -> int:
-        return len(self.__data)
+    def __validate_identity_fields(self):
+        invalid_identity_fields = []
 
-    def id(self) -> int:
+        for field in self.__identity_fields:
+            if field not in self.__data:
+                invalid_identity_fields.append(field)
+
+        if invalid_identity_fields:
+            raise InvalidIdentityFieldException(
+                f"One or more invalid identity fields specified: {invalid_identity_fields}"
+            )
+
+    def __generate_id(self) -> int:
         """
-        Return a hashed identifier for this context
+        Generates and return a hashed identifier for this context
 
         The identifier should be stable for a given context contents.
         Identifiers are used to determine rollout groups deterministically
         and consistently.
         """
-        keys = self.__data.keys()
+        keys = list(self.__identity_fields)
         vector = []
         for key in sorted(keys):
             vector.append(key)
             vector.append(str(self.__data[key]))
         hashed = hashlib.sha1(":".join(vector).encode("utf8"))
         return int.from_bytes(hashed.digest(), byteorder="big")
 
+    @property
+    def id(self):
```

Comment:
```suggestion
    def id(self) -> int:
```

You should be able to define the return of a property and have the typings workout.